cloudinit/net/__init__.py

```diff
@@ -984,6 +984,15 @@ def get_interfaces_by_mac_on_linux() -> dict:
                 )
                 continue
 
+            if name.startswith("cali") and mac == "ee:ee:ee:ee:ee:ee":
```

Comment:
> Is there an associated network driver here unique to calico devices which we can check to align with other approaches just above this segment? Ideally we don't have to parse device name and specific mac match here.
> 
> We'd generally be able to find that driver via `ls -l /sys/class/net/<calico_dev_name>/device/driver`.

What is wrong with the current approach?  Because I can't get the environment which the problem occurs. It's a CI environment and I don't have permission to log in. I added the print code to the cloud-init source and then obtained the information from the public build log. 

> Also can we please provide an inline comment to link referencing: https://docs.tigera.io/calico/latest/reference/faq#why-do-all-cali-interfaces-have-the-mac-address-eeeeeeeeeeee for future us. when understanding the context of calico device behavior?

The URL is too long. If I add it to a comment or note, the CI may fail.